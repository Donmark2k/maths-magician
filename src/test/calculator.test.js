import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Calculator from '../components/calculator';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Calculator />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
