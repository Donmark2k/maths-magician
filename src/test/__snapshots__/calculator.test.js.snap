// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="calculatorContainer"
>
  <h2
    className="calculatorH2"
  >
     Let's do some maths
  </h2>
  <div
    className="calculator"
  >
    <div
      className="display"
    >
      0
       
       
       
    </div>
    <button
      className="clear"
      onClick={[Function]}
      type="button"
      value="AC"
    >
      AC
    </button>
    <button
      className="plus-minus"
      onClick={[Function]}
      type="button"
      value="+/-"
    >
      +/-
    </button>
    <button
      className="percent"
      onClick={[Function]}
      type="button"
      value="%"
    >
      %
    </button>
    <button
      className="divide operator"
      onClick={[Function]}
      type="button"
      value="รท"
    >
      รท
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
      value="7"
    >
      7
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
      value="8"
    >
      8
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
      value="9"
    >
      9
    </button>
    <button
      className="multiply operator"
      onClick={[Function]}
      type="button"
      value="x"
    >
      ร
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
      value="4"
    >
      4
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
      value="5"
    >
      5
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
      value="6"
    >
      6
    </button>
    <button
      className="minus operator"
      onClick={[Function]}
      type="button"
      value="-"
    >
      -
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
      value="1"
    >
      1
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
      value="2"
    >
      2
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
      value="3"
    >
      3
    </button>
    <button
      className="plus operator"
      onClick={[Function]}
      type="button"
      value="+"
    >
      +
    </button>
    <button
      className="number zero"
      onClick={[Function]}
      type="button"
      value="0"
    >
      0
    </button>
    <button
      className="period"
      onClick={[Function]}
      type="button"
      value="."
    >
      .
    </button>
    <button
      className="equal operator"
      onClick={[Function]}
      type="button"
      value="="
    >
      =
    </button>
  </div>
</div>
`;
